src/snowflake/cli/plugins/stage/diff.py

```diff
@@ -149,12 +152,34 @@ def build_md5_map(list_stage_cursor: SnowflakeCursor) -> Dict[str, str]:
     Returns a mapping of relative stage paths to their md5sums.
     """
     return {
-        strip_stage_name(name): md5
-        for (name, size, md5, modified) in list_stage_cursor.fetchall()
+        strip_stage_name(file["name"]): file["md5"]
+        for file in list_stage_cursor.fetchall()
     }
 
 
-def stage_diff(local_path: Path, stage_fqn: str) -> DiffResult:
+def filter_from_diff(
+    diff: DiffResult, paths_to_sync: Set[str], prune: bool
+) -> DiffResult:
+    """Modifies the given diff, keeping only the provided paths. If prune is false, remote-only paths will be empty."""
+    diff.different = [i for i in diff.different if i in paths_to_sync]
+    diff.only_local = [i for i in diff.only_local if i in paths_to_sync]
+    if prune:
+        diff.only_on_stage = [i for i in diff.only_on_stage if i in paths_to_sync]
```

Comment:
you handle prune in a different location too, sounds odd to have to different places where we try to give that flag some semantics. I don't think the one here is needed, right?

Reply:
This is the deploying specific files flow. The full deploy flow doesn't run this function. I can consider moving the other check to run only on a full deploy.